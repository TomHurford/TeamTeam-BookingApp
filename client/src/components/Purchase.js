import React, { useState, useEffect } from "react";
import "../styles/Purchase.css";

const FutureTickets = () => {
  const [futureTickets, setFutureTickets] = useState([]);

  useEffect(() => {
    fetch("http://localhost:5001/purchase/future", {
        method: 'POST',
        body: JSON.stringify({}),
        headers: {
          'Content-Type': 'application/json'
        }
      })
      //present events from lowest to highest date
      .then((response) => response.json())
      .then((data) => {
        const sortedTickets = data.futureTickets.sort((a, b) =>
          new Date(a.event.date) - new Date(b.event.date)
        );
        setFutureTickets(sortedTickets);
      });
      
  }, []);

  return (
    <div data-testid='Purchase' className="page-container">
        <div className="underlay"></div>
       <hr></hr><hr></hr> 
      <h1>Upcoming Tickets</h1>
      <table>
        <thead>
          <tr>
            <th>Event Name</th>
            <th>Event Date</th>
            <th>Status</th>
            <th>Payment Method</th>
          </tr>
        </thead>
        <tbody>
          {futureTickets.map((ticket) => (
            <tr data-testid={ticket.id} key={ticket.id}>
              <td>{ticket.event.name}</td>
              <td>{new Date(ticket.event.date).toLocaleDateString()}</td>
              <td>{ticket.status}</td>
              <td>{ticket.paymentMethod}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default FutureTickets;
