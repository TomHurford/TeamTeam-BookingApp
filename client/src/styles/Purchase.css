table.purchase {
  border-collapse: collapse;
  width: 100%;
}

table.purchase td,
table.purchase th {
  border: 1px solid #ddd;
  padding: 8px;
}

table.purchase th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: left;
  background-color: #007aff;
  color: white;
}

table.purchase tr:nth-child(even) {
  background-color: #f2f2f2;
}

table.purchase tr:hover {
  background-color: #ddd;
}

button.purchase {
  background-color: #007aff;
  color: white;
  padding: 8px 16px;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

button.purchase:hover {
  background-color: #007aff;
}
