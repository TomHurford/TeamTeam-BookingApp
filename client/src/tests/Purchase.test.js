import React from 'react';
import renderer from 'react-test-renderer';
import Purchase from '../components/Purchase';

test('Purchase snapshot test', () => {
  const component = renderer.create(<Purchase />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});