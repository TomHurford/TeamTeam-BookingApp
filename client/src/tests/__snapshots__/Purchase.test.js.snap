// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purchase snapshot test 1`] = `
<div
  className="page-container"
  data-testid="Purchase"
>
  <div
    className="underlay"
  />
  <hr />
  <hr />
  <h1>
    Upcoming Tickets
  </h1>
  <table
    className="purchase"
  >
    <thead>
      <tr>
        <th>
          Event Name
        </th>
        <th>
          Event Date
        </th>
        <th>
          Status
        </th>
        <th>
          Payment Method
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
